app/controllers/async_info_controller.rb

```diff
@@ -61,4 +62,10 @@ def user_cache_key
     #{current_user&.articles_count}__
     #{current_user&.blocking_others_count}__"
   end
+
+  def navigation_links
+    # We're sending HTML over the wire hence 'render layout: false' enforces rails NOT TO look for a layout file to wrap
```

Comment:
You'll notice that I took out the query for `@navigation_links`. It was a mistake to add it as the query is actually being done in the view for caching purposes.